rewrite-java/src/main/java/org/openrewrite/java/cleanup/RenameLocalVariablesToCamelCase.java

```diff
@@ -0,0 +1,172 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.RenameVariable;
+import org.openrewrite.java.tree.J;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * This recipe converts local variables and method parameters to camel case convention.
+ * The recipe will not rename variables declared in for loop controls or catches with a single character.
+ *
+ * The first character is set to lower case and existing capital letters are preserved.
+ * Special characters that are allowed in java field names `$` and `_` are removed.
+ * If a special character is removed the next valid alpha-numeric will be capitalized.
+ *
+ * Currently, unsupported:
+ *  - The recipe will not rename variables declared in a class.
+ *  - The recipe will not rename variables if the result already exists in a class or the result will be a java reserved keyword.
+ */
+public class RenameLocalVariablesToCamelCase extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Rename local variables to camel case.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Renames local variables and method parameters to java camel case naming convention.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RenameNonCompliantNames();
+    }
+
+    private static class RenameNonCompliantNames extends JavaIsoVisitor<ExecutionContext> {
+        private final Pattern namingConvention = Pattern.compile("^[a-z][a-zA-Z0-9]*$");
+
+        @Override
+        public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionContext ctx) {
+            Map<J.VariableDeclarations.NamedVariable, String> renameVariablesMap = new LinkedHashMap<>();
+            Set<String> hasNameSet = new HashSet<>();
+
+            getCursor().putMessage("RENAME_VARIABLES_KEY", renameVariablesMap);
+            getCursor().putMessage("HAS_NAME_KEY", hasNameSet);
+            super.visitCompilationUnit(cu, ctx);
+
+            renameVariablesMap.forEach((key, value) -> {
+                if (!hasNameSet.contains(value)) {
+                    doAfterVisit(new RenameVariable<>(key, value));
+                    hasNameSet.add(value);
+                }
+            });
+            return super.visitCompilationUnit(cu, ctx);
```

Comment:
There's no need to call `super.visitCompilationUnit()` twice. It will result in everything being visited twice.

Reply:
I overlooked that, I'll return the cu at the end of the method. Thanks!